src/OpenTK/Math/Vector3.cs

```diff
@@ -651,9 +661,9 @@ public static Vector3 Cross(Vector3 left, Vector3 right)
         /// <param name="result">The cross product of the two inputs</param>
         public static void Cross(ref Vector3 left, ref Vector3 right, out Vector3 result)
         {
-            result = new Vector3(left.Y * right.Z - left.Z * right.Y,
-                left.Z * right.X - left.X * right.Z,
-                left.X * right.Y - left.Y * right.X);
+            result.X = left.Y * right.Z - left.Z * right.Y;
+            result.Y = left.Z * right.X - left.X * right.Z;
+            result.Z = left.X * right.Y - left.Y * right.X;
```

Comment:
Not a safe change - unlikely as it is, left or right could be pointing to the same Vector3 as result.

Reply:
I don't think it's the end of the world. It's a low-level operation and it's the user's responsibility to ensure they don't call the API wrong.

For normal uses, the multiplication operation is both fast enough and also idiot proof.

Reply:
I'll add a note about that to the doc string.

Reply:
How would this affect left or right? Their values aren't written, and vector equality is, as before, only a matter of componentwise value equality.

Reply:
Imagine you had:
```
Vector3 left = Vector3.UnitX;
Vector3 right = Vector3.UnitY;
Vector3.Cross(ref left, ref right, out left);
```
Inside Cross the first thing it does is set result.X (in this case to 0 because 0*0-0*1==0).
Next it sets result.Y but because left and result are pointing to the same memory we read the new value of X for left.X (which is 0) rather than the original value (which was 1).

Reply:
Ah, I see what you mean. I'm with Varon on this, then.

Reply:
Yes I think its fair to push the responsibility of this to users, the whole point of these functions is performance at the expense of ease of use. There are actually a few functions that already had this problem, should add the same doc comment to those really but I'd rather do that as part of an overall cleanup (probably with T4).